$OpenBSD: patch-WINGs_wview_c,v 1.1 2017/08/25 11:01:17 dcoppa Exp $

commit 83f0b764e607306feb7b17026a11034e5cf84785
Author: Doug Torrance <dtorrance@piedmont.edu>
Date:   Wed Mar 22 05:48:45 2017

WINGs: Set widget background pixmaps before realization.

Previously, when WMSetWidgetBackgroundPixmap() was called prior to
WMRealizeWidget(), no background pixmap was actually set.

This is because while the CWBackPixmap bit is correctly set to 1, the
CWBackPixel bit remains set to 1.  When XCreateWindow() is finally
called during realization, the background pixel takes precendence over the
background pixmap.

We fix this by setting CWBackPixel to 0 when setting CWBackPixmap to 1 and
vice versa.

Index: WINGs/wview.c
--- WINGs/wview.c.orig
+++ WINGs/wview.c
@@ -490,6 +490,7 @@ void W_SetViewBackgroundColor(W_View * view, WMColor *
 	view->backColor = WMRetainColor(color);
 
 	view->attribFlags |= CWBackPixel;
+	view->attribFlags &= ~CWBackPixmap;
 	view->attribs.background_pixel = W_PIXEL(color);
 	if (view->flags.realized) {
 		XSetWindowBackground(view->screen->display, view->window, W_PIXEL(color));
@@ -504,6 +505,7 @@ void W_SetViewBackgroundPixmap(W_View *view, WMPixmap 
        view->backImage = WMRetainPixmap(pix);
 
        view->attribFlags |= CWBackPixmap;
+       view->attribFlags &= ~CWBackPixel;
        view->attribs.background_pixmap = pix->pixmap;
        if (view->flags.realized) {
                XSetWindowBackgroundPixmap(view->screen->display, view->window, pix->pixmap);
